src/repo/secret/infisical.go

```diff
@@ -0,0 +1,38 @@
+package secret
+
+import (
+	"sync"
+
+	infisical "github.com/infisical/go-sdk"
+)
+
+type Secret struct {
+	client infisical.InfisicalClientInterface
+	mu     sync.Mutex
+}
+
+func NewSecret(secret *Secret) (*Secret, error) {
+	secret.mu.Lock()
+	client := infisical.NewInfisicalClient(infisical.Config{
+		SiteUrl: "https://app.infisical.com",
+	})
+	// use INFISICAL_UNIVERSAL_AUTH_CLIENT_ID, INFISICAL_UNIVERSAL_AUTH_CLIENT_SECRET env
+	_, err := client.Auth().UniversalAuthLogin("", "")
+	if err != nil {
+		return nil, err
+	}
+	secret.client = client
+	defer secret.mu.Unlock()
```

Comment:
this defer must be right after lock.
otherwise you return on line 22 and it will be locked forever